plugins/modules/swim_workflow_manager.py

```diff
@@ -752,30 +876,89 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
             else:
                 device_series_name = ".*" + device_series_name + ".*"
 
-        site_params = {
-            "site_id": site_id,
-            "device_family": device_family
-        }
+        if self.dnac_version <= self.version_2_3_5_3:
+            site_params = {
+                "site_id": site_id,
+                "device_family": device_family
+            }
 
-        try:
-            response = self.dnac._exec(
-                family="sites",
-                function='get_membership',
-                op_modifies=True,
-                params=site_params,
-            )
-        except Exception as e:
-            self.log("Unable to fetch the device(s) associated to the site '{0}' due to '{1}'".format(site_name, str(e)), "WARNING")
-            return device_uuid_list
+            try:
+                response = self.dnac._exec(
+                    family="sites",
+                    function='get_membership',
+                    op_modifies=True,
+                    params=site_params,
+                )
 
-        self.log("Received API response from 'get_membership': {0}".format(str(response)), "DEBUG")
-        response = response['device']
+            except Exception as e:
+                self.log("Unable to fetch the device(s) associated to the site '{0}' due to '{1}'".format(site_name, str(e)), "WARNING")
+                return device_uuid_list
 
-        site_response_list = []
-        for item in response:
-            if item['response']:
-                for item_dict in item['response']:
-                    site_response_list.append(item_dict)
+            self.log("Received API response from 'get_membership': {0}".format(str(response)), "DEBUG")
+            response = response['device']
+
+            site_response_list = []
+            for item in response:
+                if item['response']:
+                    for item_dict in item['response']:
+                        site_response_list.append(item_dict)
+            self.log("family = getmembership version 2.3.5.3")
+        else:
+            self.msg = (
+                "'An exception occurred: In this version '{0}' 'get_membership' is not supported'".format(self.payload.get("dnac_version")))
+            self.log(self.msg, "ERROR")
+            self.module.fail_json(msg=self.msg)
+
+        if self.dnac_version >= self.version_2_3_7_6:
+            self.log("inside the version 2.3.7.6")
+            try:
+                response = self.dnac._exec(
+                    family="site_design",
+                    function='get_site_assigned_network_devices',
+                    op_modifies=True,
+                    params={"site_id": site_id},
+                )
+                self.log("get_site_assigned_network_device")
+                self.log(response)
```

Comment:
Use one line log message only with log level.
```
self.log("Received API response from 'get_site_assigned_network_devices': {0}".format(str(response)), "DEBUG")

Reply:
updated the code